src/components/Search.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable react/prefer-stateless-function */
+import React, { ChangeEvent } from 'react';
+
+interface SearchState {
+  data: string;
+}
+
+class Search extends React.Component<object, SearchState> {
+  constructor(props: SearchState) {
+    super(props);
+    this.state = {
+      data: '',
+    };
+    this.onChange = this.onChange.bind(this);
+  }
+
+  componentDidMount() {
+    if (localStorage.searchData) this.setState({ data: localStorage.searchData });
+  }
+
+  componentWillUnmount() {
+    const { data } = this.state;
+    localStorage.searchData = data;
+  }
+
+  onChange(event: ChangeEvent<HTMLInputElement>) {
```

Comment:
it would be better if you use arrow functions instead of using bind. it makes code easier to read

  onChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { target } = event;
    this.setState({ data: target.value });
  }

Reply:
Solved in last task